commit 5218f2f94d01fa74519472d0d72e9c43e9792624
Author: Matthew Farrellee <matt@redhat.com>
Date:   Wed Feb 25 13:32:43 2009 -0600

    Port to gcc 4.4's stricter string functions
    
    gcc 4.4 is now enforcing constness for strchr strrchr strstr etc. This
    means when we pass in a const char * we need to take out a const char
    *, not a char *.

diff --git a/src/condor_amazon/gsoap_commands.cpp b/src/condor_amazon/gsoap_commands.cpp
index 064096d..514adab 100644
--- a/src/condor_amazon/gsoap_commands.cpp
+++ b/src/condor_amazon/gsoap_commands.cpp
@@ -97,7 +97,7 @@ AmazonRequest::ParseSoapError(const char* callerstring)
 
 		// NOTE: The faultcode appears to have a qualifying 
 		// namespace, which we need to strip
-		char *s = strrchr(*code, ':');
+		const char *s = strrchr(*code, ':');
 		if( s ) {
 			s++;
 			if( !strncasecmp(s, "Client.", strlen("Client.")) ) {
diff --git a/src/condor_c++_util/store_cred.cpp b/src/condor_c++_util/store_cred.cpp
index 9d43fc8..49da998 100644
--- a/src/condor_c++_util/store_cred.cpp
+++ b/src/condor_c++_util/store_cred.cpp
@@ -170,7 +170,7 @@ char* getStoredCredential(const char *username, const char *domain)
 
 int store_cred_service(const char *user, const char *pw, int mode)
 {
-	char *at = strchr(user, '@');
+	const char *at = strchr(user, '@');
 	if ((at == NULL) || (at == user)) {
 		dprintf(D_ALWAYS, "store_cred: malformed user name\n");
 		return FAILURE;
diff --git a/src/condor_gridmanager/gt42_gahp_wrapper.cpp b/src/condor_gridmanager/gt42_gahp_wrapper.cpp
index 63762ab..0336c80 100644
--- a/src/condor_gridmanager/gt42_gahp_wrapper.cpp
+++ b/src/condor_gridmanager/gt42_gahp_wrapper.cpp
@@ -160,7 +160,7 @@ main( int argc, char* argv[] ) {
 	Directory dir( buff.Value() );
 	dir.Rewind();
 	while ( (ctmp = dir.Next()) ) {
-		char *match = strstr( ctmp, ".jar" );
+		const char *match = strstr( ctmp, ".jar" );
 		if ( match && strlen( match ) == 4 ) {
 			classpath += classpath_seperator;
 			classpath += dir.GetFullPath();
@@ -170,7 +170,7 @@ main( int argc, char* argv[] ) {
 	Directory dir2( buff.Value() );
 	dir2.Rewind();
 	while ( (ctmp = dir2.Next()) ) {
-		char *match = strstr( ctmp, ".jar" );
+		const char *match = strstr( ctmp, ".jar" );
 		if ( match && strlen( match ) == 4 ) {
 			classpath += classpath_seperator;
 			classpath += dir2.GetFullPath();
diff --git a/src/condor_gridmanager/gt4_gahp_wrapper.cpp b/src/condor_gridmanager/gt4_gahp_wrapper.cpp
index c9f23ed..3d1c09b 100644
--- a/src/condor_gridmanager/gt4_gahp_wrapper.cpp
+++ b/src/condor_gridmanager/gt4_gahp_wrapper.cpp
@@ -160,7 +160,7 @@ main( int argc, char* argv[] ) {
 	Directory dir( buff.Value() );
 	dir.Rewind();
 	while ( (ctmp = dir.Next()) ) {
-		char *match = strstr( ctmp, ".jar" );
+		const char *match = strstr( ctmp, ".jar" );
 		if ( match && strlen( match ) == 4 ) {
 			classpath += classpath_seperator;
 			classpath += dir.GetFullPath();
